[22.01] Fix FormElement: doesn't populate options for parameters

davelopez opened the pull request:
Fixes #13597

Remove `text` as the default type for Form input elements, otherwise when the form input defines `options` they won't be rendered as a select and will be forced to a regular text field.

## How to test the changes?
- [x] Instructions for manual testing are as follows:
  - Follow the steps in #13597

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

davelopez committed:
Remove `text` as default type for Form input elements

Otherwise when the input defines `options` they won't be rendered as a select and will be forced to a regular text field

guerler reviewed (approved):
This is the correct fix, thank you. The reason being is that in `parameters.js` we match unspecified types to `text` or `select` and that the quota inputs in the admin controller come without explicitly specified types. We should add a client test either here or in a follow-up in dev to cover for this.

For future reference:
- https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/controllers/admin.py#L734
- https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Form/Elements/parameters.js#L39

davelopez committed:
Add test to check which type of input will be rendered

When no type is explicitly set in the input definition it will default to normal text input, if the definition contains `options` it will be a select input
